asterius/src/Asterius/TypesConv.hs

```diff
@@ -22,23 +21,19 @@ import System.IO
     withFile,
   )
 
-{-# INLINE marshalToModuleSymbol #-}
-marshalToModuleSymbol :: GHC.Module -> AsteriusModuleSymbol
-marshalToModuleSymbol (GHC.Module u m) = AsteriusModuleSymbol
-  { unitId = GHC.fs_bs $ GHC.unitIdFS u,
-    moduleName =
-      CBS.splitWith (== '.') $ GHC.fs_bs $ GHC.moduleNameFS m
-  }
-
-{-# INLINE zEncodeModuleSymbol #-}
-zEncodeModuleSymbol :: AsteriusModuleSymbol -> String
-zEncodeModuleSymbol AsteriusModuleSymbol {..} =
+{-# INLINE zEncodeModule #-}
+zEncodeModule :: GHC.Module -> String
```

Comment:
We can obtain the unit id part as `unitId = GHC.unitIdFS m`, module name part as `moduleName = GHC.moduleNameFS m`, so we can simply concat these two as `FastString`: `unitId <> "_" <> moduleName`, avoiding redundant computations.

Reply:
Ah, right. I was too hasty with this but it should be fixed now.